.github/workflows/bin-ci.yml

```
@@ -96,7 +96,7 @@ jobs:
           --junitfile "$TEST_RESULTS_DIR/${{ matrix.consul-version }}/gotestsum-report.xml" -- \
           -race "${PACKAGE_NAMES[@]}" \
           -- "$FLAG"
-    - uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3.1.2
+    - uses: actions/upload-artifact@v4
```

We should pin the GitHub Action for better security.
```suggestion
    - uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 #v4.6.2
```